tools/community/list_collabs.py

```diff
@@ -0,0 +1,77 @@
+"""
+Usage: list_collabs.py [--user USER] [--token TOKEN]
+
+Create a list of all collaborators that commited to an organization. Provide user name and access token, if you need to
+send many requests.
+
+  -h --help         show this
+  -u --user USER    github user name
+  -t --token TOKEN  github token (see https://docs.github.com/en/rest/overview/other-authentication-methods#basic-authentication)
+"""
+from docopt import docopt
+import requests
+from requests.auth import HTTPBasicAuth
+
+if __name__ == '__main__':
+    organization = "precice"
+    arguments = docopt(__doc__)
+
+    commit_authors = set()
+    skipped_commits = []
+    number_commits = 0
+
+    # Collect all repos under organization
+    repos = set()
+    api_url = "https://api.github.com/users/{}/repos?per_page=100".format(organization)
+    if '--user' in arguments.keys():
+        response = requests.get(api_url, auth=HTTPBasicAuth(arguments['--user'], arguments['--token']))
+    else:
+        response = requests.get(api_url)
+
+    try:
+        if 'message' in response.json().keys():
+            print(response.json()['message'])
+            quit()
+    except AttributeError:
+        pass
+
+    for repo in response.json():
+        repos.add(repo["name"])
+
+    print("Parsing {} repos: {}".format(len(repos), repos))
+
+    # Traverse all repos
+    for repo in repos:
+        print("Parsing {}...".format(repo))
+        pageno = 1
+        api_url = "https://api.github.com/repos/{}/{}/commits?per_page=100&page={}".format(organization, repo, pageno)
+
+        # Use authentication, if provided
+        if '--user' in arguments.keys():
+            response = requests.get(api_url, auth=HTTPBasicAuth(arguments['--user'], arguments['--token']))
+        else:
+            response = requests.get(api_url)
+
+        # While page has entries, continue
+        while len(response.json()) > 0:
+            print("On page {} got {} hits.".format(pageno, len(response.json())))
+            pageno += 1
+            api_url = "https://api.github.com/repos/{}/{}/commits?per_page=100&page={}".format(organization, repo, pageno)
+            if '--user' in arguments.keys():
+                response = requests.get(api_url, auth=HTTPBasicAuth(arguments['--user'], arguments['--token']))
+            else:
+                response = requests.get(api_url)
```

Comment:
We should also check the response for being valid. When running without a user/password, I got stuck in a "On page (increasing) got 2 hits.", but no error. I had to guess that this is related to the access.